pkg/distributor/otlp/cmd/generate/main.go

```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package main
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/grafana/regexp"
+
+	"github.com/google/go-github/v57/github"
+)
+
+const (
+	dpath = "storage/remote/otlptranslator/prometheusremotewrite"
+	owner = "grafana"
+	repo  = "mimir-prometheus"
+)
+
+func main() {
+	app := kingpin.New("generate", "Generate OTLP translation code.")
+	checkFlag := app.Flag("check", "Verify that generated OTLP translation code corresponds to mimir-prometheus version.").Bool()
+
+	generate := func(*kingpin.ParseContext) error {
+		sourceRefPath := filepath.Join("cmd", "generate", ".source-ref")
+
+		if *checkFlag {
+			return check(sourceRefPath)
+		}
+
+		sources, sourceRef, err := downloadSources(sourceRefPath)
```

Comment:
I'm not sure I understand why we need to download the sources? If we put a reference to the sources anywhere (make a test/dummy command), they would be downloaded by `go mod vendor` wouldn't they?

Reply:
That's a good idea @krajorama - my thinking was that these sources wouldn't be vendored. I guess it's preferable to rely on vendoring.

Reply:
Done.